File: Longest-Consecutive-Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {

        unordered_set<int> st(nums.begin(),nums.end());

        int ans = 0;

        for(int num:st){
            if(st.find(num-1)==st.end()){
                int len =1;
                int n = num;

                while(st.find(n+1)!= st.end()){
                    n++;
                    len++;
                }

                ans = max(ans,len);
            }

        }

        return ans;
        
    }
};

File: Longest-Consecutive-Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        st = set(nums)
        res = 0

        for n in st:

            if n-1 not in st:
                len = 1

                while n+1 in st:
                    n = n+1
                    len+=1

                res = max(res,len)        

        return res